src/modules/csm/state.py

```diff
@@ -102,7 +104,16 @@ def add_processed_epoch(self, epoch: EpochNumber) -> None:
     def log_progress(self) -> None:
         logger.info({"msg": f"Processed {len(self._processed_epochs)} of {len(self._epochs_to_process)} epochs"})
 
-    def migrate(self, l_epoch: EpochNumber, r_epoch: EpochNumber):
+    def migrate(self, l_epoch: EpochNumber, r_epoch: EpochNumber, consensus_version: int):
+        if consensus_version != self._consensus_version:
+            logger.warning(
+                {
+                    "msg": f"Cache was built for consensus version {self._consensus_version}. "
+                    f"Discarding data to migrate to consensus version {consensus_version}"
+                }
+            )
+            self.clear()
```

Comment:
Maybe rename clear to `discard_collected_data`?

Reply:
`clear` is fine because the function clears all the state.